File: Novomber 2022/Day 17/C++ Code/Count of Subarrays.cpp
// Problem Link :- https://practice.geeksforgeeks.org/problems/count-of-subarrays5922/1
// Video Solution Link :- https://youtu.be/cluFktVZPhA

//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

#define ll long long


// } Driver Code Ends
//User function template for C++
class Solution{
public:
	// #define ll long long

// 	Time Complexity :- O(n);
// 	Space Complexity :- O(1);

	ll countSubarray(int arr[], int n, int k) {
	    // code here
	    ll ans = (ll)n*(n+1)/2;
        ll l =0;
        for(int i=0;i<n;i++) {
            if(arr[i]<=k) {
                l++;
            } else {
                ans-=(l*(l+1)/2);
                l=0;
            }
        }
        ans-=(l*(l+1)/2);
        return ans;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, k, i;
        cin >> n >> k;
        int arr[n];
        for (i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.countSubarray(arr, n, k);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Novomber 2022/Day 17/JAVA Code/Count of Subarrays.java
// Problem Link :- https://practice.geeksforgeeks.org/problems/count-of-subarrays5922/1
// Video Solution Link :- https://youtu.be/cluFktVZPhA

//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            String[] str = br.readLine().trim().split(" ");
            int n = Integer.parseInt(str[0]);
            int k = Integer.parseInt(str[1]);
            str = br.readLine().trim().split(" ");
            int[] arr = new int[n];
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(str[i]);
            }

            long ans = new Solution().countSubarray(arr, n, k);

            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    // 	Time Complexity :- O(n);
    // 	Space Complexity :- O(1);
    
    long countSubarray(int arr[], int n, int k) {

        // code here
        long ans = (long)n*(n+1)/2;
        long l =0;
        for(int ar : arr) {
            if(ar<=k) {
                l++;
            } else {
                ans-=(l*(l+1)/2);
                l=0;
            }
        }
        ans-=(l*(l+1)/2);
        return ans;
    }
}
